pkg/client/client.go

```diff
@@ -346,14 +346,27 @@ func (c *Client) CreateOrUpdateSecurityContextConstraints(ctx context.Context, s
 	return errors.Wrap(err, "updating SecurityContextConstraints object failed")
 }
 
-func (c *Client) CreateRouteIfNotExists(ctx context.Context, r *routev1.Route) error {
+func (c *Client) CreateOrUpdateRoute(ctx context.Context, r *routev1.Route) error {
 	rclient := c.osrclient.RouteV1().Routes(r.GetNamespace())
-	_, err := rclient.Get(ctx, r.GetName(), metav1.GetOptions{})
+	existing, err := rclient.Get(ctx, r.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		_, err := rclient.Create(ctx, r, metav1.CreateOptions{})
 		return errors.Wrap(err, "creating Route object failed")
 	}
-	return nil
+	if err != nil {
+		return errors.Wrap(err, "retrieving Route object failed")
+	}
+	if reflect.DeepEqual(r.Spec, existing.Spec) {
```

Comment:
I think that the `spec.host` will always be different because it's automatically chosen if unset.

Reply:
@simonpasquier is [correct](https://github.com/openshift/api/blob/a99ffa1cac6709edf8f502b16890b16f9a557e00/route/v1/types.go#L75-L80).

Dropping this code path probably makes most sense?